### PR TITLE
Exclude test projects from source-build

### PR DESCRIPTION
**Pull request by dseefeld**

### CODE REVIEW

File: src/PerformanceTests/Utilities/ReferenceAssemblies/ReferenceAssemblies.csproj [hunk 1]
```diff
@@ -4,6 +4,7 @@
     <TargetFramework>net6.0</TargetFramework>
     <IsPackable>false</IsPackable>
     <NonShipping>true</NonShipping>
+    <ExcludeFromSourceBuild>true</ExcludeFromSourceBuild>
```
Comment (Evangelink) [inline src/PerformanceTests/Utilities/ReferenceAssemblies/ReferenceAssemblies.csproj:7]: Shall it be defined as described by @sharwell in https://github.com/dotnet/arcade/issues/7132#issue-837193193?

Comment (dseefeld) [reply src/PerformanceTests/Utilities/ReferenceAssemblies/ReferenceAssemblies.csproj:7]: I don't think so.  @sharwell is suggesting a change to arcade to make this easier for repos.

**Review by Youssef1313 — COMMENTED** [review]
Why not in `Directory.Build.props` with a `'$(NonShipping) == 'true'` condition?

**Comment by dseefeld** [comment]
> Why not in `Directory.Build.props` with a `'$(NonShipping)' == 'true'` condition?

As noted in the arcade comment above:

  > Some of the tools we use in the repository are needed for the build to complete but don't actually ship anywhere.

**Review by sharwell — APPROVED** [review]

**Pull request merged**